tests/subsys/debug/assert_custom_header fails in multiple platforms

### Describe the bug

tests/subsys/debug/assert_custom_header fails to build in multiple (33) platforms


### Regression

- [x] This is a regression.

### Steps to reproduce

1. $ mkdir build && cd build
2. $ cmake -GNinja -DBOARD=nrf5340dk/nrf5340/cpuapp/ns ../tests/subsys/debug/assert_custom_header/
3. $ ninja
4. see error

### Relevant log output

```shell
In file included from build/tfm/../zephyr/misc/generated/syscalls_links/include/zephyr/irq_multilevel.h:15,
                 from build/tfm/../zephyr/misc/generated/syscalls_links/include/zephyr/devicetree.h:20,
                 from modules/trusted-firmware-m/nordic/include/RTE_Device.h:14,
                 from modules/tee/tf-m/trusted-firmware-m/platform/ext/target/nordic_nrf/common/core/./nrfx_config.h:35,
                 from modules/hal/nordic/nrfx/nrfx.h:37,
                 from modules/tee/tf-m/trusted-firmware-m/platform/ext/target/nordic_nrf/common/core/common/cmsis.h:35,
                 from modules/tee/tf-m/trusted-firmware-m/platform/include/tfm_hal_device_header.h:14,
                 from modules/tee/tf-m/trusted-firmware-m/bl2/ext/mcuboot/bl2_main.c:21:
build/tfm/../zephyr/misc/generated/syscalls_links/include/zephyr/sys/__assert.h:154:10: fatal error: zephyr_custom_assert.h: No such file or directory
  154 | #include <zephyr_custom_assert.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
```
33 failures (33 targets fail to build it) in weekly run
https://github.com/zephyrproject-rtos/zephyr/actions/runs/19997797171 
For ex.:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/19997797171/job/57348087573#step:14:6231

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: Any
- Zephyr SDK 0.17.4
- SHA main since ada3aad811d2f74ff50a48843797858c334dffa3

### Additional Context

Test added in https://github.com/zephyrproject-rtos/zephyr/pull/96514
Fails since it was introduced

--- a/tests/subsys/debug/assert_custom_header/testcase.yaml
+++ b/tests/subsys/debug/assert_custom_header/testcase.yaml
@@ -1,6 +1,11 @@
 common:
   integration_platforms:
     - native_sim
+  platform_allow:
+    - native_sim
+    - native_sim/native/64
+    - qemu_x86
+    - qemu_cortex_m3
 
 tests:
   debug.assert_custom_header:

